### ISSUE TITLE
Background geolocation v3 [urgent]

### ISSUE THREAD

**Issue by rodrigoords** [ionic-team/ionic-native#2775]
[ ] bug report
[ X ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://ionicworldwide.herokuapp.com/

**Current behavior:**
v2 version does not work properly on new Android version:
https://github.com/mauron85/cordova-plugin-background-geolocation/issues/493#issuecomment-429156163

**Expected behavior:**
Implement v3 plugin

**Comment by pavankumarkatakam** [comment]
Since the cordova plugin background geolocation will not work in Oreo when the package point at 2.X branch. Please make changes and point at 3.0 alpha branch

**Comment by VeerababuKottu** [comment]
Our app also needs V3 of Cordova plugin of background geolocation

**Comment by rodrigoords** [comment]
No, i tried to use Capacitor to create interface but without success.

**Comment by pavankumarkatakam** [comment]
Im directly using the BackgroundGeoLocation Plugin with out ionic, it can be accessible via window object.

**Comment by rodrigoords** [comment]
Can you give some code example plz

**Comment by pavankumarkatakam** [comment]
Sorry, it will not came under window object you can directly access the variable, if typescript shows that an error just declare it globally
`declare var BackgroundGeolocation: any;`

And use it as normal as specified in the documentation like
`BackgroundGeolocation.start();`

Also no need to put this in the construtor or ngOnInit

**Comment by rodrigoords** [comment]
Work like a charm... thanks

**Comment by devsaran** [comment]
I get error..

`Uncaught Error: Module cordova-plugin-mauron85-background-geolocation.BackgroundGeolocation does not exist.`

Where should I put 

`declare var BackgroundGeolocation: any;`

**Comment by rajesh99** [comment]
I am using ionic 4.  I get the following error
Uncaught Error: Module cordova-plugin-mauron85-background-geolocation.BackgroundGeolocation does not exist

**Comment by malavshah9** [comment]
Uncaught Error: Module cordova-plugin-mauron85-background-geolocation.BackgroundGeolocation does not exist
I am getting same error. If any solution, please provide.

**Comment by bitforcesrl** [comment]
same here with ionic 4 the weird part is that with android live debug works, in production mode not... i have that error, and now the sorce maps are not working on ionic 4

**Comment by TaylorHawkes** [comment]
I had this issue when upgrading from version 2 to 3. 

My issue was platforms/android/platform_www/plugins/cordova-plugin-mauron85-background-geolocation was not getting removed when I removed the plugin. 

I manually removed that folder, and re-installed v3 of the plugin and error went away. Think something went wonky on my Mac, because I still had a the "www/backgroundGeolocation.js" and it should have been uppercase "B" - BackgroundGeolocation.js.
